app/controllers/appropriate_bodies/participants_controller.rb

```diff
@@ -5,9 +5,11 @@
 module AppropriateBodies
   class ParticipantsController < BaseController
     def index
-      collection = InductionRecordsQuery.new(appropriate_body:).induction_records
+      induction_records = InductionRecordsQuery.new(appropriate_body:).induction_records
+      participant_profiles = induction_records.map(&:participant_profile)
```

Comment:
@ethax-ross Seems like we have a N+1 query in this line..
We are fetching all participant profiles for all induction records before sending it to the filters and to pagy to only display 50 records per page in the end.
Can we somehow fetch participant profile and other data for only those 50 records in the first place?

Reply:
hmm yeah there is an N+1 here, I'll see if I can address that. We can't filter/apply pagination first unfortunately because the training states are used _in_ the filters 😬